crypto/fipsmodule/CMakeLists.txt

```diff
@@ -84,6 +124,39 @@ if(ARCH STREQUAL "aarch64")
     sha512-armv8.${ASM_EXT}
     vpaes-armv8.${ASM_EXT}
   )
+
+  if (UNIX AND NOT APPLE)
```

Comment:
Looks rather duplicated, with above. Is there an easy way to refactor this?

Reply:
I'll try to come up with some simplification

Reply:
I refactored the code to avoid duplication, please see the new commit.

Reply:
Neat solution.